amespahdbpythonsuite/mcfitted.py

```diff
@@ -0,0 +1,360 @@
+#!/usr/bin/env python4
+
+from __future__ import annotations
+from typing import Optional
+
+import numpy as np
+import matplotlib.pyplot as plt  # type: ignore
+
+from astropy import units as u  # type: ignore
+from scipy import stats  # type: ignore
+from specutils import Spectrum1D  # type: ignore
+
+from amespahdbpythonsuite.amespahdb import AmesPAHdb
+
+message = AmesPAHdb.message
+
+
+class MCFitted:
+    """
+    AmesPAHdbPythonSuite mcfitted class.
+    Contains methods to fit and plot the input spectrum using a Monte Carlo approach.
+
+    """
+
+    def __init__(self, d: Optional[dict] = None, **keywords) -> None:
+        self.__set(d, **keywords)
+
+    def __set(self, d: Optional[dict] = None, **keywords) -> None:
+        """
+        Populate data dictionary helper.
+
+        """
+        self.mcfits = keywords.get("mcfits", list())
+        self.distribution = keywords.get("distribution", "")
+
+        if isinstance(d, dict):
+            if d.get("type", "") == self.__class__.__name__:
+                if "mcfits" not in keywords:
+                    self.mcfits = d["mcfits"]
+                if "distribution" not in keywords:
+                    self.distribuion = d["distribuion"]
+
+    def get(self) -> dict:
+        """
+        Calls class: :class:`amespahdbpythonsuite.transitions.Transitions.get`.
+        Assigns class variables from inherited dictionary.
+
+        """
+        d = {}
+        d["type"] = self.__class__.__name__
+        d["mcfits"] = self.mcfits
+        d["distribution"] = self.distribution
+
+        return d
+
+    def _getstats(self, d=dict()) -> dict:
```

Comment:
Possibly this function is not needed, as it is basically a single line:

```
stat = {'mean': stats.describe(d).mean, 'std': np.sqrt(stats.describe(d).variance), 'skew': stats.describe(d).skewness, 'kurt': stats.describe(d).kurtosis}
```

Reply:
I made this function given that we already have one-line functions (e.g. getnorm, getfit, getresidual etc in fitted.py) and I am actually calling it 5 times in mcfitted.py, and may be potentially used in future instances. However If you think this is better to keep it in each function I can very well do that.

Reply:
At some point we'll need to do a code-review and can consider things like this. For now, its just a small detail.